static/src/javascripts/projects/commercial/modules/close-disabled-slots.js

```diff
@@ -1,23 +1,38 @@
 // @flow
 import qwery from 'qwery';
 import fastdom from 'lib/fastdom-promise';
+import { commercialFeatures } from 'commercial/modules/commercial-features';
 
 const adSlotSelector: string = '.js-ad-slot';
+const mpuCandidateClass: string = 'fc-slice__item--mpu-candidate';
+const mpuHiderClass: string = 'fc-slice__item--no-mpu';
+const mpuCandidateSelector: string = `.${mpuCandidateClass}`;
+
+const shouldDisableAdSlotWhenAdFree = adSlot =>
+    commercialFeatures.adFree &&
+    (adSlot.className.toLowerCase().includes(mpuCandidateClass) ||
+        !adSlot.className.toLowerCase().includes('merchandising'));
 
 const shouldDisableAdSlot = adSlot =>
-    window.getComputedStyle(adSlot).display === 'none';
+    window.getComputedStyle(adSlot).display === 'none' ||
+    shouldDisableAdSlotWhenAdFree(adSlot);
 
 const closeDisabledSlots = (force: boolean): Promise<void> => {
     // Get all ad slots
     let adSlots: Array<Element> = qwery(adSlotSelector);
+    let mpuCandidates: Array<Element> = qwery(mpuCandidateSelector);
 
     if (!force) {
         // remove the ones which should not be there
         adSlots = adSlots.filter(shouldDisableAdSlot);
+        mpuCandidates = mpuCandidates.filter(shouldDisableAdSlot);
```

Comment:
Maybe these could avoid the filter by using the whole list if it is `adFree` rather than mixing into the rules for reading the styles and classes from the adSlot? It will always check the `display` in this case when it doesn't need to... (Or maybe reorder the OR logic)